include/dfu/dfu_target_full_modem.h

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+/** @file dfu_target_full_modem.h
+ *
+ * @defgroup dfu_target_full_modem Full Modem DFU Target
+ * @{
+ * @brief DFU Target full modem updates.
+ */
+
+#ifndef DFU_TARGET_FULL_MODEM_H__
+#define DFU_TARGET_FULL_MODEM_H__
+
+#include <stddef.h>
+#include <sys/types.h>
+#include <device.h>
+#include <dfu/dfu_target.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct dfu_target_fmfu_fdev {
+	/* Flash device to use for storing the modem firmware. */
+	const struct device *dev;
+
+	/* Offset within the device to store the modem firmware. */
+	off_t offset;
+
+	/* Number of bytes to set aside in the device for storing the modem
+	 * firmware. Set this to '0' to use all available flash (starting from
+	 * `offset`.
+	 */
+	size_t size;
+};
+
+struct dfu_target_full_modem_params {
+	/* Buffer for writes to `fmfu_dev`. */
+	uint8_t *buf;
+
+	/* Length of `buf` */
+	size_t len;
+
+	/* Flash device for storing the modem firmware */
+	struct dfu_target_fmfu_fdev *dev;
+};
+
+/**
+ * @brief Configure resources required by dfu_target_full_modem
+ *
+ * @param[in] params Pointer to dfu target full modem update parameters.
+ *
+ * @retval non-negative integer if successful, otherwise negative errno.
+ */
+int dfu_target_full_modem_cfg(struct dfu_target_full_modem_params *params);
+
+/**
+ * @brief See if data in buf indicates a full modem update.
+ *
+ * @retval true if data matches, false otherwise.
+ */
+bool dfu_target_full_modem_identify(const void *const buf);
+
+/**
+ * @brief Initialize dfu target, perform steps necessary to receive firmware.
+ *
+ * @param[in] file_size Size of the current file being downloaded.
```

Comment:
`callback` is undocumented.

Reply:
Its not used, so I was not quite sure how to explain it. Please check the proposal.